Two phase (almost)

sariths opened the pull request:
@mostaphaRoudsari , there is that issue with NROWs that I have run up against. I think you had addressed it once on the forum here: http://radiance-online.org/pipermail/radiance-general/2015-February/010754.html. 

The recipe/work flow is in test.py

mostaphaRoudsari commented:
@sariths I assign this to you so you can merge it in once you feel it's ready. Let me know if there is anything that I need to check or help.

sariths commented:
Combined notes for this commit: 

1. Added check in commanbase.py to allow inputFiles that are not lists/tuples.
2. The input weaFile for gendaymtx will be handled through `@property` rather than a descriptor.
3. Started genskyvec and rcollate, still WIP.
4. Coded rfluxmtx, added `classmethod` to create skies on the fly and also to check for `#@rfluxmtx` type strings on other senders/receivers.
5. Added a bare-bones implementation of rmtxop. Which can handle two phase method and probably three-phase also unless we try fancy stuff like specifying individual options for matrices. 
6. Added a readonly descriptor for datatype based on https://github.com/ladybug-analysis-tools/honeybee/issues/65
7. Added specular sampling options to defaultset.py
8. Added lw, lr and ss options to gridbased class.
9. test.py will run a two-phase calc.

Notes:
1. I will be adding comments/unit-tests after some of these scripts  after some more testing. I feel it is an overkill to do the documentation right now and then change it over and over (which I have done already !). 
2.

sariths commented:
Hi @mostaphaRoudsari , can you run the test.py on your system and see if you can get a full run going? It works fine on my workstation and laptop.

sariths commented:
@mostaphaRoudsari There is a bug in the way ambient accuracy is set. The script tries to convert the floating value to int everytime although that isn't the correct way. 

> UserWarning: The expected type for aa (ambient accuracy) is <type 'int'>.The provided input 0.1 has been converted to 0

I am just noting this here to highlight it as a a known issue.

mostaphaRoudsari commented:
@sariths I finally get a chance to try to run the test file. It fails on my system to generate .wea file under temp and because of that gendaymtx also fails. I will try to figure it out and report back.

sariths commented:
@mostaphaRoudsari What is the error that you get ?

mostaphaRoudsari commented:
It was failing to create the wea file. I needed to create a temp file under tests\room.

mostaphaRoudsari commented:
@sariths after [adding radiance\bin and radiance\lib](https://github.com/ladybug-analysis-tools/honeybee/issues/68) it works like a charm! Nice work! :100:

sariths commented:
@mostaphaRoudsari :) Let's wait for the fireworks till we start validating results with Daysim ! I think we have somewhat of a foundation for the workflow on which we can build rest of the features.

mostaphaRoudsari commented:
@sariths I have the daysim analysis running for the last hour or so and it is still running. I set the parameters to similar values as yours for the comparative study but maybe I should just set it up to a high quality that can at least finish at some point.

mostaphaRoudsari commented:
@sariths check [this file](https://github.com/ladybug-analysis-tools/honeybee/files/275673/comparative_annual_analysis_results.xlsx)
 for comparative results. The pattern is similar but the values from Daysim is higher. I ran the analysis for couple of hours using gendaylit and the results is similar to results from daysim. I ran daysim with default honeybee parameters for now.

I also have question. How did you set up the values for combination parameter? Is it for converting rgb values to illuminance?

sariths commented:
@mostaphaRoudsari Finally, things are getting interesting ! I am traveling tomorrow and shall be in transit, settling down etc. till Sunday. I should be able to get to this some time on Sunday.

mostaphaRoudsari commented:
@sariths thanks! Safe travels/move. I will be working on ladybug for the next couple of days. 2-Phase looks very close to be done.

sariths commented:
@mostaphaRoudsari , Quick question... how long does it take for the 2-phase routine to run on your system ? On my system it is a few seconds. To me, that in itself is a red flag as daysim takes a few hours to do the same stuff...
